sourcecode/weather_records_calculation.py

```diff
@@ -0,0 +1,161 @@
+import calendar
+from datetime import datetime
+
+from .weather_records_reader import WeatherRecordsReader
+
+
+class WeatherRecordsCalculation:
+    def __init__(self) -> None:
+        """Constructor for initializing reports"""
+        self.year_report = {
+            'highest_temp': float('-inf'),
+            'highest_temp_date': "",
+            'lowest_temp_date': "",
+            'lowest_temp': float('inf'),
+            'humidity': float('-inf'),
+            'humidity_date': "",
+        }
+
+        self.month_report = {
+            'avg_highest_temp': float('-inf'),
+            'avg_lowest_temp': float('inf'),
+            'avg_mean_humidity': float('-inf'),
+        }
+
+    def calculate_year_record(self, weather_files: list) -> dict:
+        """
+        Calculates Highest Temperature, Lowest Temperature
+        and Humidity.
+
+        @params
+        :weather_files list: hold list of files path.
+
+        @return
+        :dict: contains report data
+        """
+        for weather_file in weather_files:
+            weather_records = WeatherRecordsReader.weather_info(weather_file)
+
+            for weather_day_info in weather_records:                
+                self.max_temperature(weather_day_info)
+                self.min_temperature(weather_day_info)
+                self.max_humidity(weather_day_info)
+
+        return self.year_report
+
+    def calculate_month_record(self, weather_file: str) -> dict:
+        """
+        Calculates Highest Temperature, Lowest Temperature
+        and Humidity.
+
+        @params
+        :weather_file str: hold file path.
+        
+        @return
+        :dict: contains report data
+        """
+        total_max_temp = 0
+        total_min_temp = 0
+        total_mean_humidity = 0
+
+        weather_records = WeatherRecordsReader.weather_info(weather_file)    
+        for weather_day_info in weather_records:
+            if weather_day_info['Max TemperatureC'] != "":
+                total_max_temp += int(weather_day_info['Max TemperatureC'])
```

Comment:
Can we avoid if condition by using the code below:
total_max_temp += int(weather_day_info['Max TemperatureC'] or '0')